java/test/jglome/GlomeTest.java

```diff
@@ -0,0 +1,153 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jglome;
+
+import static java.security.spec.NamedParameterSpec.X25519;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static jglome.TestVector.testVector1;
+import static jglome.TestVector.testVector2;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.XECPrivateKey;
+import java.security.interfaces.XECPublicKey;
+import java.util.Arrays;
+import java.util.Optional;
+import jglome.Glome.WrongTagException;
+import org.junit.jupiter.api.Test;
+import jglome.Glome.CounterOutOfBoundsException;
+import jglome.Glome.MinPeerTagLengthOutOfBoundsException;
+import jglome.Glome.StaticSecureRandom;
+
+/**
+ * Class, which contains tests for main.jglome.Glome functionality.
+ */
+public class GlomeTest {
+
+  final static int N_TEST_VECTORS = 2;
+
+  TestVector[] testVectors = {testVector1, testVector2};
+  Glome[][] glomeManagers = new Glome[N_TEST_VECTORS][2]; // first is for A, second is for B
+  KeyPair[] aKeys = new KeyPair[N_TEST_VECTORS];
+  KeyPair[] bKeys = new KeyPair[N_TEST_VECTORS];
+
+  GlomeTest() throws MinPeerTagLengthOutOfBoundsException {
+    for (int i = 0; i < N_TEST_VECTORS; i++) {
+      aKeys[i] = deriveKeysFromPrivate(testVectors[i].kah);
+      bKeys[i] = deriveKeysFromPrivate(testVectors[i].kbh);
+      glomeManagers[i][0] = new Glome((XECPublicKey) bKeys[i].getPublic(),
+          (XECPrivateKey) aKeys[i].getPrivate(), 32);
+      glomeManagers[i][1] = new Glome((XECPublicKey) aKeys[i].getPublic(),
+          (XECPrivateKey) bKeys[i].getPrivate(), 28);
```

Comment:
Maybe store 28 in a constant that is to be used also in `testShouldFailWhenIncorrectTruncatedTag` and `testCorrectTruncatedTag`.